deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/base/Utils.kt

```diff
@@ -6,6 +6,41 @@ import com.airbnb.deeplinkdispatch.configurablePathSegmentSuffix
 import com.airbnb.deeplinkdispatch.configurablePathSegmentPrefix
 import java.io.InputStream
 
+/**
+ * Chunk a CharSequence based on how long it's Modified UTF-8
+ * (https://en.wikipedia.org/wiki/UTF-8#Modified_UTF-8) ByteArray representaiton would be.
+ */
+fun CharSequence.chunkOnModifiedUtf8ByteSize(chunkSize: Int): List<CharSequence> {
+    require (chunkSize >= 3) {
+        "UTF-8 chars can be up to 3 bytes wide. Minumum chunk size is 3 bytes."
+    }
+    var modifiedUtf8BlockSize = 0
```

Comment:
I think this and the variable on L25 should be named consistently to each other so they're easier to associate, potentially `next(Chunk|Block)ModifiedUtf8Size`

Reply:
Good catch about the terminology, i did not even notice I used two.